examples/task_creation.py

```diff
@@ -0,0 +1,86 @@
+"""
+Example of Task creation
+"""
+
+from typing import Literal
+
+from pydantic import Field, ValidationInfo, field_validator
+
+import aind_behavior_curriculum as abc
+
+
+class ExampleTaskParameters(abc.TaskParameters):
+    """
+    Example Task Parameters
+    """
+
+    # Required: Define type annotations for strict type checks.
+    # Make fields immutable with Literal type.
+    field_1: int = Field(default=0, ge=0.0)
+    field_2: int = Field(default=0, ge=0.0)
+    field_3: float = Field(default=0.5, ge=0.0, le=1.0)
+    field_4: float = Field(default=0.5, ge=0.0, le=1.0)
+    field_5: Literal["Immutable Field"] = "Immutable Field"
+
+    # Optional: Add additional validation to fields.
+    @field_validator("field_1", "field_2")
+    @classmethod
+    def check_something(cls, v: int, info: ValidationInfo):
+        """Your validation code here"""
+        return v
+
+
+class ExampleTask(abc.Task):
+    """
+    Example Task
+    """
+
+    name: Literal["TaskName"] = "TaskName"
+    description: str = Field(default="Example description of task")
+    version: abc.SemVerAnnotation = abc.__version__
```

Comment:
For clarity, maybe dont use the `abc.__version__` since that is what you ask people NOT to do in the comment below.